test/steps/kogitoapp.go

```diff
@@ -53,42 +52,32 @@ func registerKogitoAppSteps(s *godog.Suite, data *Data) {
 }
 
 // Deploy service steps
-func (data *Data) deployQuarkusExampleServiceWithConfiguration(contextDir string, table *gherkin.DataTable) error {
-	kogitoApp, err := getKogitoAppExamplesStub(data.Namespace, contextDir, table)
-	if err != nil {
-		return err
-	}
-
-	kogitoApp.Spec.Runtime = v1alpha1.QuarkusRuntimeType
-
-	return framework.DeployService(data.Namespace, framework.GetDefaultInstallerType(), kogitoApp)
-}
 
-func (data *Data) deploySpringBootExampleServiceWithConfiguration(contextDir string, table *gherkin.DataTable) error {
+func (data *Data) deployExampleServiceWithConfiguration(runtimeType, contextDir string, table *gherkin.DataTable) error {
 	kogitoApp, err := getKogitoAppExamplesStub(data.Namespace, contextDir, table)
 	if err != nil {
 		return err
 	}
-	if kogitoApp.Spec.Build.Native {
-		return fmt.Errorf("Spring Boot does not support native build")
+	if runtimeType != "quarkus" && kogitoApp.Spec.Build.Native {
```

Comment:
I would parse first and then do the check, this will avoid the string

Reply:
changed